flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/BackendRestorerProcedure.java

```diff
@@ -105,6 +105,8 @@ public T createAndRestore(@Nonnull List<? extends Collection<S>> restoreOptions)
 
 			++alternativeIdx;
 
+			// IMPORTANT: please be careful when modifying the log statements because they are used for validation in
+			// the automatic end-to-end tests. Those tests might fail if they are not aligned with the log message!
 			if (restoreState.isEmpty()) {
 				LOG.debug("Creating {} with empty state.", logDescription);
 			} else {
```

Comment:
nit: since this log's format is so special, maybe we can introduce a `static final Sting LOG_FORMAT = "Creating {} with empty state."` for it, and guard it by adding a unit test (feel free to ignore).

Reply:
I feel that guarding a log statement format by a test, just so that another test won't fail is a bit too much. I mean, it does not really break anything so a comment should be ok.